src/lib/optimizer/join_graph_builder.cpp

```diff
@@ -0,0 +1,265 @@
+#include "join_graph_builder.hpp"
```

Comment:
This is mostly the JoinGraphBuilder that was already part of the master prior to #882, simplified now thanks to the new expression system. A thorough review shouldn't be required.